File: sdk/samples/WCFSamples/TechnologySamples/Basic/WorkflowServices/StateMachineWorkflowService/CS/Service/StateMachineCalculatorService.xoml.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace Microsoft.WorkflowServices.Samples
{
	public partial class StateMachineCalculatorService: StateMachineWorkflowActivity
	{
        [NonSerialized]
        public int inputValue = default(int);
        public int currentValue = default(int);
        
        private void Add(object sender, EventArgs e)
        {
            currentValue += inputValue;
        }

        private void Subtract(object sender, EventArgs e)
        {
            currentValue -= inputValue;
        }

        private void Multiply(object sender, EventArgs e)
        {
            currentValue *= inputValue;
        }

        private void Divide(object sender, EventArgs e)
        {
            currentValue /= inputValue;
        }
	}

}

File: sdk/samples/WCFSamples/TechnologySamples/Basic/WorkflowServices/StateMachineWorkflowService/VB/Service/StateMachineCalculatorService.xoml.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections
Imports System.Drawing
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

Namespace Microsoft.WorkflowServices.Samples
    <Serializable()> _
    Partial Public Class StateMachineCalculatorService
        Inherits StateMachineWorkflowActivity
        <NonSerialized()> _
        Public inputValue As Integer = Nothing
        Public currentValue As Integer = Nothing

        Private Sub Add(ByVal sender As Object, ByVal e As EventArgs)
            currentValue += inputValue
        End Sub

        Private Sub Subtract(ByVal sender As Object, ByVal e As EventArgs)
            currentValue -= inputValue
        End Sub

        Private Sub Multiply(ByVal sender As Object, ByVal e As EventArgs)
            currentValue *= inputValue
        End Sub

        Private Sub Divide(ByVal sender As Object, ByVal e As EventArgs)
            currentValue /= inputValue
        End Sub
    End Class

End Namespace